src/input_common.rs

```diff
@@ -111,6 +111,7 @@ pub enum ReadlineCmd {
     DisableMouseTracking,
     // ncurses uses the obvious name
     ClearScreenAndRepaint,
+    MoveJumpAnchor,
```

Comment:
I think this is definitely a problem we should consider addressing.

The classic use case is if you want to edit `NEEDLE` after pasting this command:

	echo '/some/really/long/path/name with spaces/aaa/bbb/ccc' NEEDLE 'other token with spaces' a/b/c/

Today I usually fall back on Alt+e to open my editor.

Possible alternative approaches are
- use Shift+Left and Shift+Right which are bound to `forward-bigword` and `backward-bigword` by default -- I didn't realize that until today
- allow searching within the commandline (Ctrl-S would be the obvious Emacs-binding)
- allow moving by full tokens (the Alt+Shift+b binding will be open once we support CSI u)
- add commands that move by `$COLUMNS/2` or similar

Your approach sounds reasonable as well and the UI is fairly well known and obvious.
Do you have an example use case?

I briefly tried it and it has some glitches on multiline commands (when using \co twice) and the contrast is bad (light theme).

Reply:
Thanks for having a look.

I tried using it on multiline commands, and it does not seem to behave in any unexpected way to me, even when pressing ctrl+o twice, however i do agree that the behaviour probably is undesirable, especially replacing the newlines with jump letters, and thus temporarily altering the line breaking.  

There are also some challenges with regards to the fact that it skips adding jump characters if a word is too short leading to inconsistent marking of characters depending on the location of the cursor.

I haven't really used this feature much in the context of fish yet, so I am not sure how well it fits for command line editing except for jumping in paths as i mentioned, but as I wrote i find it useful in the context of a programming text editor; for example when jumping in a line of code that consists of an  an if-test with lots of brackets, quotes etc.

When it comes to the highlighting colour I agree that the contrast is poor, i just chose the first sensible colour enumeration in the theme I could find. We can change it to something else, or make a completely new color category.